Add notifications to downloader.py

engrbm87 opened the pull request:
## Description:
Fire an event when download is completed or failed.

**Related issue (if applicable):** fixes #<home-assistant issue number goes here>

**Pull request in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io) with documentation (if applicable):** home-assistant/home-assistant.github.io#4916

## Example entry for `configuration.yaml` (if applicable):
```yaml

```

## Checklist:
  - [x] The code change is tested and works locally.

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

If the code communicates with devices, web services, or third-party tools:
  - [ ] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [ ] New dependencies have been added to the `REQUIREMENTS` variable ([example][ex-requir]).
  - [ ] New dependencies are only imported inside functions that use them ([example][ex-import]).
  - [ ] New dependencies have been added to `requirements_all.txt` by running `script/gen_requirements_all.py`.
  - [ ] New files were added to `.coveragerc`.

If the code does not interact with devices:
  - [ ] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**
  - [ ] Tests have been added to verify that the new code works.

[ex-requir]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L14
[ex-import]: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/keyboard.py#L54

engrbm87 committed:
Update downloader.py

Add persistent notification to alert when download is finished or in case of download failure.

engrbm87 committed:
Update downloader.py

engrbm87 committed:
Update downloader.py

engrbm87 committed:
Update downloader.py

balloob reviewed (changes requested):
This is the wrong approach. Instead, fire events so that people can choose to create a notification via an automation or flash lights etc.

engrbm87 committed:
Fire and event when download is requested

Added 2 events to represent download completed and download failed. This will allow the user to trigger an automation based on the status of the download.

engrbm87 committed:
Update downloader.py

balloob commented on homeassistant/components/downloader.py:
```diff
@@ -133,9 +135,19 @@ def do_download():
 
                     _LOGGER.debug("Downloading of %s done", url)
+                    hass.bus.fire(
+                         "{}.{}".format(DOMAIN, DOWNLOAD_COMPLETED_EVENT), {
```
We don't have event names with periods in it, please use `_`

engrbm87 committed:
Update downloader.py

replaced . with _

balloob commented:
Ok to merge when a PR for the docs is included and linting errors have been resolved

engrbm87 committed:
Update downloader.py

fixed linting errors

balloob commented:
Awesome, thanks!
